usdigitalresponse/usdr-gost
# [issue-2369] Back or Browse Grants Link from Grant Details

### Back or Browse Grants Link from Grant Details [issue-2369](https://github.com/usdigitalresponse/usdr-gost/issues/2369) 

## Description
This PR implements navigation links in the Grant Details page as per the provided design specifications and user navigation scenarios.

## Screenshots / Demo Video

https://github.com/usdigitalresponse/usdr-gost/assets/485547/b6c3c124-d544-4dca-984c-1416a3112f48

### Testing
1. Verify Navigation from Grants Table to Grant Details:
    - Click on a grant item in the Grants Table component.
    - Ensure that you are redirected to the Grant Details page.
    - Confirm that the URL includes the grant ID as a parameter.

2. Check Back Link Behavior:
    - Visit the Grant Details page from various parts of the application (browse grants and my grant).
    - Click on the "Back" link above the grant title.
    - Verify that you are redirected to the previous page you were on.
    - If the previous page was saved search results, confirm that you return to the specific saved search results.
    - If the previous page was a tab within the My Grants page, ensure that you return to the same tab.
    - Compare the behavior with clicking the browser's back button.


### Automated and Unit Tests

### Manual tests for Reviewer

## Checklist
